plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -2108,7 +2108,7 @@ private void finishOptimize(ConnectorSession session, DeltaLakeTableExecuteHandl
 
             for (String scannedPath : scannedPaths) {
                 String relativePath = relativePath(tableLocation, scannedPath);
-                transactionLogWriter.appendRemoveFileEntry(new RemoveFileEntry(relativePath, writeTimestamp, false));
+                transactionLogWriter.appendRemoveFileEntry(new RemoveFileEntry(toUriFormat(relativePath), writeTimestamp, false));
```

Comment:
Which test cover this change?

Reply:
I've extended the existing `testOptimize` test to have rows with partition values `with:colon`.